libs/nss.c

```diff
@@ -11,7 +11,16 @@
 
 #include "tcb.h"
 
+/* readdir_r(3) is deprecated since glibc 2.24 */
+#if defined __GLIBC_PREREQ && __GLIBC_PREREQ(2, 24)
+#define GLIBC_2_24 1
```

Comment:
I suggest to introduce `USE_READDIR` instead of `GLIBC_2_24`.
Also I suggest to make this macro defined to either 0 or 1, this results to more clear code.

Reply:
Fixed in rebased commit.